src/snowflake/snowpark/stored_procedure_profiler.py

```diff
@@ -0,0 +1,128 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+import re
+import threading
+from typing import List, Literal, Optional
+
+import snowflake.snowpark
+from snowflake.snowpark._internal.utils import validate_object_name
+
+STORED_PROCEDURE_CALL_PATTERN = r"WITH\s+.*?\s+AS\s+PROCEDURE\s+.*?\s+CALL\s+.*"
```

Comment:
just found out we have `SNOWFLAKE_ANONYMOUS_CALL_WITH_PATTERN` in `snowflake/snowpark/_internal/utils.py`. let's use that.